From f13970e138bee543c160a879337d5354d3479b20 Mon Sep 17 00:00:00 2001
From: Weixin Zhou <zwx@rock-chips.com>
Date: Sat, 22 Dec 2018 14:42:54 +0800
Subject: [PATCH] arm64: dts: rockchip: set sdmmc power off in suspend for
 rk3399pro evb

The vccio_sd keepon in suspend will causes 10ma power consumption to
increase if there is no serial cable connection.

Set vccio_sd and vcc_sd power off in suspend for rk3399pro evb v10/v11.

Change-Id: I935de4028bea5d03a0397b81bd63c99757155b4d
Signed-off-by: Weixin Zhou <zwx@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10.dts |  6 ++----
 arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11.dts | 14 ++++++--------
 2 files changed, 8 insertions(+), 12 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10.dts b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10.dts
index e3b0b6de60e6..e707c48c8923 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10.dts
@@ -569,8 +569,7 @@
 
 				regulator-name = "vccio_sd";
 				regulator-state-mem {
-					regulator-on-in-suspend;
-					regulator-suspend-microvolt = <3300000>;
+					regulator-off-in-suspend;
 				};
 			};
 
@@ -582,8 +581,7 @@
 
 				regulator-name = "vcc_sd";
 				regulator-state-mem {
-					regulator-on-in-suspend;
-					regulator-suspend-microvolt = <3300000>;
+					regulator-off-in-suspend;
 				};
 			};
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11.dts b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11.dts
index 20efb7e4a597..8eed6934757a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11.dts
@@ -345,15 +345,15 @@
 	status = "okay";
 };
 
+&hdmi_in_vopb {
+	status = "disabled";
+};
+
 &i2s2 {
 	#sound-dai-cells = <0>;
 	status = "okay";
 };
 
-&hdmi_in_vopb {
-	status = "disabled";
-};
-
 &i2c0 {
 	status = "okay";
 	i2c-scl-rising-time-ns = <180>;
@@ -575,8 +575,7 @@
 
 				regulator-name = "vccio_sd";
 				regulator-state-mem {
-					regulator-on-in-suspend;
-					regulator-suspend-microvolt = <3300000>;
+					regulator-off-in-suspend;
 				};
 			};
 
@@ -588,8 +587,7 @@
 
 				regulator-name = "vcc_sd";
 				regulator-state-mem {
-					regulator-on-in-suspend;
-					regulator-suspend-microvolt = <3300000>;
+					regulator-off-in-suspend;
 				};
 			};
 
-- 
2.35.3

